crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -1,187 +1,191 @@
-    // The test assures the mapper publishes smartrest messages 114 and 500 on `c8y/s/us` which shall be send over to the cloud if bridge connection exists.
-    let broker = mqtt_tests::test_mqtt_broker();
-    let mut messages = broker.messages_published_on("c8y/s/us").await;
-
+async fn mapper_publishes_init_messages_on_startup() {
     // Start SM Mapper
     let cfg_dir = TempTedgeDir::new();
-    let (_tmp_dir, sm_mapper) = start_c8y_mapper(broker.port, &cfg_dir).await.unwrap();
+    let (mqtt, _http, _fs, _timer) = spawn_c8y_mapper_actor(&cfg_dir, true).await;
 
-    // Expect 500 messages has been received on `c8y/s/us`, if no msg received for the timeout the test fails.
-    mqtt_tests::assert_received_all_expected(&mut messages, TEST_TIMEOUT_MS, &["500\n"]).await;
+    let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
 
-    sm_mapper.abort();
+    let version = env!("CARGO_PKG_VERSION");
+    let default_fragment_content = json!({
+        "c8y_Agent": {
+            "name": "thin-edge.io",
+            "url": "https://thin-edge.io",
+            "version": version
+        }
+    })
+    .to_string();
+
+    assert_received_contains_str(
+        &mut mqtt,
+        [
+            (
+                "c8y/inventory/managedObjects/update/test-device",
+                default_fragment_content.as_str(),
+            ),
+            ("c8y/s/us", "114"),
+            (
+                "c8y/inventory/managedObjects/update/test-device",
+                &json!({"type":"test-device-type"}).to_string(),
+            ),
+            ("c8y/s/us", "500"),
+            ("tedge/commands/req/software/list", r#"{"id":"#),
+        ],
+    )
+    .await;
 }
 
-#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+#[tokio::test]
 #[serial]
 async fn mapper_publishes_software_update_request() {
     // The test assures SM Mapper correctly receives software update request smartrest message on `c8y/s/ds`
     // and converts it to thin-edge json message published on `tedge/commands/req/software/update`.
-    let broker = mqtt_tests::test_mqtt_broker();
-    let mut messages = broker
-        .messages_published_on("tedge/commands/req/software/update")
-        .await;
-    let cfg_dir = TempTedgeDir::new();
-    let (_tmp_dir, sm_mapper) = start_c8y_mapper(broker.port, &cfg_dir).await.unwrap();
-    // Prepare and publish a software update smartrest request on `c8y/s/ds`.
-    let smartrest = r#"528,test-device,nodered,1.0.0::debian,,install"#;
-    broker.publish("c8y/s/ds", smartrest).await.unwrap();
-    publish_a_fake_jwt_token(broker).await;
-
-    let expected_update_list = r#"
-         "updateList": [
-            {
-                "type": "debian",
-                "modules": [
+    let (mqtt, http, _fs, _timer) = spawn_c8y_mapper_actor(&TempTedgeDir::new(), true).await;
+    spawn_dummy_c8y_http_proxy(http);
+
+    let mut mqtt = mqtt.with_timeout(TEST_TIMEOUT_MS);
+
```

Comment:
I think better to explain which messages are skipped.

Reply:
I'll add a comment. It's there in some tests, but missed in a few others.